pkg/codegen/go/gen.go

```diff
@@ -1757,6 +1786,8 @@ func generatePackageContextMap(tool string, pkg *schema.Package, goInfo GoPackag
 		pkg.resources = append(pkg.resources, r)
 
 		pkg.names.Add(resourceName(r))
+		pkg.names.Add(resourceName(r) + "Input")
+		pkg.names.Add(resourceName(r) + "Output")
```

Comment:
These two lines are probably the most controversial since these will rename types that have already been emitted. However, in all such cases the code is unable to be compiled due to a redeclaration, e.g. for AWS:
```
# github.com/pulumi/pulumi-aws/sdk/v3/go/aws/kinesis
go/aws/kinesis/analyticsApplication.go:274:43: impossible type assertion:
	AnalyticsApplicationOutput does not implement pulumi.Output (missing Apply method)
go/aws/kinesis/pulumiTypes.go:2204:6: AnalyticsApplicationOutput redeclared in this block
	previous declaration at go/aws/kinesis/analyticsApplication.go:356:6
```

As a result, these don't violate our breaking change policy.

Reply:
Do you think it's problematic that we rely on the ordering of resources in the schema? Would it be better to sort (probably in multiple places here) to get more predictability?

Reply:
> Do you think it's problematic that we rely on the ordering of resources in the schema? Would it be better to sort (probably in multiple places here) to get more predictability?

Thanks for the pointer. In general I think we want to be as deterministic as possible. I will follow up shortly with sorted version (I have to do something similar for collisions with input/output variants among types themselves.